### DISCUSSION TITLE
Problem with Content Security Policy

### DISCUSSION THREAD

**Discussion by gabrielgrs** [vercel/next.js#70034] [Help]
### Summary

Hello! I’ve been struggling for days to implement a Content Security Policy with a nonce, but I haven’t been successful. Every time I generate the nonce and try to access the platform, my console throws dozens of errors, preventing me from using the tool.

I am using App Router in version "14.2.5", with a custom node server because my application is hosted on Azure.
I already tried using middleware.ts but the "Content-Security-Policy" was not show in the network request!

Errors generated in the console
![CleanShot 2024-09-12 at 12 58 13@2x](https://github.com/user-attachments/assets/8d89cb59-ed71-4604-9bfd-9ce2a14ccf8b)

The result in the network
![CleanShot 2024-09-12 at 12 58 45@2x](https://github.com/user-attachments/assets/52a8ea45-bfdb-4338-87fb-be60a2e32bc9)

The implementation in next.config.mjs
![CleanShot 2024-09-12 at 13 01 48@2x](https://github.com/user-attachments/assets/638e717e-5a5b-4461-9c96-c193a502c4ec)

**Comment by Juneezee** [comment]
Your CSP is blocking your own JavaScript assets. See https://nextjs.org/docs/app/building-your-application/configuring/content-security-policy#adding-a-nonce-with-middleware

> We recommend ignoring matching prefetches (from `next/link`) and static assets that don't need the CSP header.
